database-plugins/src/main/java/co/cask/hydrator/plugin/db/batch/source/DBSource.java

```diff
@@ -158,39 +104,26 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
    * we get the message and throw it as BadRequestException
    */
   @Path("getSchema")
-  public Map<String, String> getSchema(GetSchemaRequest request,
+  public Schema getSchema(GetSchemaRequest request,
```

Comment:
Schema will not get serialized correctly unless the Gson instance used to do it has the SchemaTypeAdapter registered.  So we either need to return Schema.toString() here, or make a change in cdap to support the Schema object as a return type.